### ISSUE TITLE
Communication Errors

### ISSUE THREAD

**Issue by CKHX** [d4rken/jt-dpm8600-psu-mqtt#5]
Hi!

at first I'm a total "newbe" so, sorry for asking perhaps stupid thinks...
I've got the Juntek DPM8650 (TTL Version) and try to connect an esp board like the picture in this Projekt.
I've compiled the software (my first time do this with Visual Studio Code and platformio but i think it works :-)
The esp communicate with my MQTT Broker (iobroker) and i recieve the topics.
The "TXD" led on the ttl Adapter is flasching, also the "com" led on the DPM.
But... there is no "RXD" flasching.

the Terminal output:
### START: updateStats ###
Voltage: -10.00
Current: -11,00
Max current: -15,00
Power: -12,00
Temp: -16,00
### END: updateStats ###

as i understand the DPM ist not answering (reading Error)? I'm correct?
is there some special Settings to do at the DPM?
I try on the DPM "5-CS __1_" and "5-CS __0_" both with the same result...

Ore... is it the wrong Version of DPM, do i have to use the RS485 Verion?

Thanks for your Help!

Greetings 
Chris

**Comment by d4rken** [comment]
I don't think you need the RS485 version, you can probably use the TTL version if you wire it directly to the ESP. I don't know why it's not working though you'll have to figure that out for yourself.

I would try change more settings (mode? baud rate?) on the DPM or try to add some debug statement in the communication library to see if anything useful is received at all.

**Comment by CKHX** [comment]
Hi!
I connect it now directly, with the same result. 
I try several settings on DPM, every Setting with the same result.
As I#m a "Newbee" im "fighting" with platformio do get some debbug results, at the moment there is no debuging.

during compiling i get some warnings...
`Compiling .pio\build\d1_mini\lib463\SPI\SPI.cpp.o
.pio\libdeps\d1_mini\OneWire\OneWire.cpp: In member function 'uint8_t OneWire::reset()':
.pio\libdeps\d1_mini\OneWire\OneWire.cpp:167:24: warning: unused variable 'reg' [-Wunused-variable]
  167 |  volatile IO_REG_TYPE *reg IO_REG_BASE_ATTR = baseReg;
      |                        ^~~
.pio\libdeps\d1_mini\OneWire\OneWire.cpp: In member function 'void OneWire::write_bit(uint8_t)':
.pio\libdeps\d1_mini\OneWire\OneWire.cpp:201:24: warning: unused variable 'reg' [-Wunused-variable]
  201 |  volatile IO_REG_TYPE *reg IO_REG_BASE_ATTR = baseReg;
      |                        ^~~
.pio\libdeps\d1_mini\OneWire\OneWire.cpp: In member function 'uint8_t OneWire::read_bit()':
.pio\libdeps\d1_mini\OneWire\OneWire.cpp:229:24: warning: unused variable 'reg' [-Wunused-variable]
  229 |  volatile IO_REG_TYPE *reg IO_REG_BASE_ATTR = baseReg;
      |                        ^~~
Archiving .pio\build\d1_mini\lib109\libOneWire.a`

`Compiling .pio\build\d1_mini\lib33e\ESP8266WiFi\BearSSLHelpers.cpp.o
.pio\libdeps\d1_mini\PubSubClient\src\PubSubClient.cpp: In member function 'boolean PubSubClient::publish_P(const char*, const uint8_t*, unsigned int, boolean)':
.pio\libdeps\d1_mini\PubSubClient\src\PubSubClient.cpp:523:16: warning: comparison of integer expressions of different signedness: 'unsigned int' and 'int' [-Wsign-compare]
  523 |     return (rc == expectedLength);
      |             ~~~^~~~~~~~~~~~~~~~~
Archiving .pio\build\d1_mini\libd57\libPubSubClient.a`

So I think I need more time to find th eProblem 
where can I find the settings for the baud rate of ESP ?
on DPM i try severell...

**Comment by mhin76** [comment]
Hi Chris

I have the same behavior. the transmit flashes, but it never gets any read. have you solved the issue? Strange is that I do not see any info in my MQTT server. Ah, just found the bug. the MQTT server IP needs to be filled into the MQTT_SERVER and not the MQTT_SERVER_IP in the config file!
I am using the D1 mini (USB C version).
Kind regards,
Martin

**Comment by cn72** [comment]
Hi,
for all who have a problem with the communications. Begin with 2400 Baud. I have here a Wemos D1mini that did not work with 4800 or 9600 Baud. 3 Day later I try 2400 and it now work perfect.
Greetings
Christoph
